pkgs/development/node-packages/default.nix

```diff
@@ -35,6 +35,25 @@ let
       '';
     };
 
+    hyperspace-cli = super."@hyperspace/cli".override {
+      nativeBuildInputs = [
+          pkgs.makeWrapper
+          pkgs.libtool
+          pkgs.autoconf
+          pkgs.automake
+      ];
```

Comment:
```suggestion
      nativeBuildInputs = with pkgs; [
        makeWrapper
        libtool
        autoconf
        automake
      ];
```

Reply:
Is it not arguable that avoiding the use of `with` here is beneficial as it won't have to be evaluated, or something like that?

Reply:
i don't like the use of with, adam say it should, in his words 'die in a fire'. Unless someone can say why with is a better approach to code readability, or not..

@SuperSandro2000 , whats you scope on this, why with , apart from the 16 ascii bytes it saves in the commit?

Reply:
The reason I didn't use `with` is because of what I've learned here. https://nix.dev/anti-patterns/language#with-attrset-expression

Reply:
We are repeating the same word 4 times in a list. Pretty good use of with. You are just not supposed to use it over the entire scope.

Reply:
> This maybe cool kid deep nix lang blackmagic that i don't understand.

No, languages over 20 years ago already had with.

> whats you scope on this, why with , apart from the 16 ascii bytes it saves in the commit?

To not repeat pkgs 4 times. I don't care about the bytes saved. also this is in UTF-8.

Reply:
@SuperSandro2000 
> Static analysis can’t reason about the code, because it would have to actually evaluate this file to see what variables are in scope.

This is the only reason I have not to use with here, is this somehow not a valid criticism of `with` in this context? If everything in Nix stopped using `with` in this way that you're proposing, wouldn't it be better?

Reply:
Is there even a static analyzer for nix? Also I am pretty sure that we are doing enough things which need evaluation anyway.

> If everything in nixpkgs stopped using with in this way that you're proposing, wouldn't it be better on a technical level?

No because we would need to repeat a long of things everywhere and that is not the goal.